### PR TITLE
Adds support for NMEA 0183 MDA sentence type

### PR DESCRIPTION
**Pull request by thorsteinssonh**
Adds support for NMEA 0183 MDA sentence type, containing meteorological composite.
This message type is used by Thies Clima, and probably other major instrument makers.

Changes to be committed:
	modified:   README.md
	new file:   hooks/MDA.js
	modified:   hooks/index.js
	new file:   test/MDA.js

### CODE REVIEW

**Comment by tkurki** [comment]
Thanks! This is mergeable, but I suggest two changes.

You could make it a little bit more elegant / less verbose if you construct the `values` array directly as you go through the fields instead of setting variables conditionally and then later on having another null condition check on what you just set a few lines earlier.

Like this:

```
  const values = []
  ...
  if( parts[4] != '' ) {
    values.push({
      path: 'environment.outside.temperature',
      value: utils.transform(utils.float(parts[4]), 'c', 'k')
    })
  }
  ...
  const delta = {
    updates: [
      {
        source: tags.source,
        timestamp: tags.timestamp,
        values
      }
    ],
  }
```

Another nitpick would be to change `!=` to `!==`,  which is a good practice to do always in Javascript, even if it makes no difference here. See https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

**Comment by thorsteinssonh** [comment]
Thanks tkurki and sbender9.
Ok I have pushed the improvements suggested by tkurki.
Note that this MDA message type can result in duplicate outside.pressure and wind.speedOverGround, depending on how the manufacturer uses the message.

It seems best just to allow those duplicates, if the manufacturer puts that into the message, even though they should contain exact same values.

Reason for the duplicates is that MDA definition is rather clumbsy, allowing wind speed and pressure entries in differen units. In signalk this just gets converted to SI units.

File: hooks/MDA.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  *  12.    Checksum
+  */
+
+module.exports = function (input) {
+  const { id, sentence, parts, tags } = input
+
+  const values = []
+
+  // make SI units override any non-SI units
+  let pressure = null
```
Comment (tkurki) [inline hooks/MDA.js]: Unused variable here

Comment (thorsteinssonh) [reply hooks/MDA.js]: oopps, fixed.
I should use a linter.

**Comment by tkurki** [comment]
Well, we really should have a precommit hook to run lint.

Thank you very much!

**Pull request merged**